tgui/packages/tgui/interfaces/MatMarket.js

```diff
@@ -0,0 +1,113 @@
+import { useBackend } from '../backend';
+import { Section, Stack, Flex, Button } from '../components';
+import { Window } from '../layouts';
+
+export const MatMarket = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const { materials = [] } = data;
```

Comment:
```suggestion
  const { act, data } = useBackend<Data>(context);   // this will tell your editor that data is the type listed above
  
  const { orderingPrive, canOrderCargo, creditBalance, materials = [] } = data; // better to destructure here (style nit)
```